test/src/testRpcClient.cpp

```diff
@@ -0,0 +1,331 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <csignal>
+#include <spdlog/spdlog.h>
+#include <up-client-zenoh-cpp/client/upZenohClient.h>
+#include <up-cpp/transport/builder/UAttributesBuilder.h>
+#include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <gtest/gtest.h>
+
+using namespace uprotocol::utransport;
+using namespace uprotocol::v1;
+using namespace uprotocol::uri;
+using namespace uprotocol::rpc;
+using namespace uprotocol::client;
+
+UUri rpcUri = LongUriSerializer::deserialize("/test_rpc.app/1/rpc.handler");
+UUri rpcNoServerUri = LongUriSerializer::deserialize("/test_rpc.app/1/rpc.noServer");
+
+class RpcServer : public UListener {
+
+     public:
+
+        UStatus onReceive(UMessage &message) const override {
+
+            UStatus status;
+
+            status.set_code(UCode::OK);
+            
+            auto builder = UAttributesBuilder::response(message.attributes().source(), message.attributes().sink(), UPriority::UPRIORITY_CS0, message.attributes().id());
+
+            UAttributes responseAttributes = builder.build();
+
+            UPayload outPayload = message.payload();
+
+            UMessage respMessage(outPayload, responseAttributes);
+
+            if (nullptr != message.payload().data()) {
+
+                std::string cmd(message.payload().data(), message.payload().data() + message.payload().size());
+
+                if ("No Response" != cmd) {
+                    return UpZenohClient::instance()->send(respMessage);
+                }
+            } else {
+                return UpZenohClient::instance()->send(respMessage);
+            }
+                   
+            return status;
+        }
+};
+
+class ResponseListener : public UListener {
+
+     public:
+
+        UStatus onReceive(UMessage &message) const override {
+
+            (void) message;
+
+            UStatus status;
+
+            status.set_code(UCode::OK);
+           
+            return status;
+        }
+};
+
+class TestRPcClient : public ::testing::Test {
+
+    public:
+        // SetUpTestSuite() is called before all tests in the test suite
+        static void SetUpTestSuite() {
+
+        }
+
+        // TearDownTestSuite() is called after all tests in the test suite
+        static void TearDownTestSuite() {
+
+        }
```

Comment:
Instead of copying the setup/teardown code into each test, why not just swap `SetUpTestSuite()` and `TearDownTestSuite()` for `SetUp()` and `TearDown()`? The latter are run at the [start and end of each test](http://google.github.io/googletest/reference/testing.html#Test::SetUp). That would look something like this:

```suggestion
        // SetUp() is called before each tests in the test suite
        static void SetUp() {
            zenoh_client = UpZenohClient::instance();
            zenoh_client->registerListener(rpcUri, TestRPcClient::rpcListener);
        }

        // TearDown() is called after each tests in the test suite
        static void TearDown() {
            zenoh_client->unregisterListener(rpcUri, TestRPcClient::rpcListener);
            zenoh_client.reset();
        }
        
        static inline std::shared_ptr<UpZenohClient> zenoh_client;
```

Reply:
This seems reasonable.  I'll make the changes, but plan to put them in a PR of their own instead of continuing to add to this one.